_LowLevelCode/cpp/test/mex_bin_plugin.tests/mex_bin_plugin.test.cpp

```diff
@@ -0,0 +1,168 @@
+#include "mex_bin_plugin/bin_io_handler.h"
+#include "utility/environment.h"
+#include <gtest/gtest.h>
+
+using namespace Horace::Utility;
+
+bool file_exists(const  std::string& filename) {
+    std::ifstream f(filename.c_str());
+    return f.good();
+};
+
+void del_file(const std::string& filename) {
+    std::remove(filename.c_str());
+};
+
+
+TEST(TestMexBinPlugin, write_bin_data_no_file) {
+    const std::string horace_root{
+        Environment::get_env_variable(Environment::HORACE_ROOT, ".") };
+
+    std::string binary_file{ horace_root + "/_test/binary_write.bin" };
+    del_file(binary_file);
```

Comment:
Seems unwise to start deleting files from `.` If the environment vars are not set up correctly should error.

Reply:
yes but they are set-up properly by cmake as far as I remember.